packages/grafana-ui/src/components/CustomScrollbar/CustomScrollbar.tsx

```diff
@@ -1,142 +1,170 @@
-import React, { Component } from 'react';
+import React, { FC, useCallback, useEffect, useRef } from 'react';
 import isNil from 'lodash/isNil';
 import classNames from 'classnames';
+import { css } from 'emotion';
 import Scrollbars from 'react-custom-scrollbars';
+import { useStyles } from '../../themes';
+import { GrafanaTheme } from '@grafana/data';
 
 interface Props {
   className?: string;
   autoHide?: boolean;
   autoHideTimeout?: number;
-  autoHideDuration?: number;
   autoHeightMax?: string;
   hideTracksWhenNotNeeded?: boolean;
   hideHorizontalTrack?: boolean;
   hideVerticalTrack?: boolean;
   scrollTop?: number;
-  setScrollTop: (event: any) => void;
+  setScrollTop?: (event: any) => void;
   autoHeightMin?: number | string;
   updateAfterMountMs?: number;
 }
 
 /**
  * Wraps component into <Scrollbars> component from `react-custom-scrollbars`
  */
-export class CustomScrollbar extends Component<Props> {
-  static defaultProps: Partial<Props> = {
-    autoHide: false,
-    autoHideTimeout: 200,
-    autoHideDuration: 200,
-    setScrollTop: () => {},
-    hideTracksWhenNotNeeded: false,
-    autoHeightMin: '0',
-    autoHeightMax: '100%',
-  };
-
-  private ref: React.RefObject<Scrollbars>;
-
-  constructor(props: Props) {
-    super(props);
-    this.ref = React.createRef<Scrollbars>();
-  }
-
-  updateScroll() {
-    const ref = this.ref.current;
-    const { scrollTop } = this.props;
-
-    if (ref && !isNil(scrollTop)) {
-      ref.scrollTop(scrollTop);
-    }
-  }
-
-  componentDidMount() {
-    this.updateScroll();
-
-    // this logic is to make scrollbar visible when content is added body after mount
-    if (this.props.updateAfterMountMs) {
-      setTimeout(() => this.updateAfterMount(), this.props.updateAfterMountMs);
+export const CustomScrollbar: FC<Props> = ({
+  autoHide = false,
+  autoHideTimeout = 200,
+  setScrollTop,
+  className,
+  autoHeightMin = '0',
+  autoHeightMax = '100%',
+  hideTracksWhenNotNeeded = false,
+  hideHorizontalTrack,
+  hideVerticalTrack,
+  scrollTop,
+  updateAfterMountMs,
+  children,
+}) => {
+  const ref = useRef<Scrollbars>(null);
+  const styles = useStyles(getStyles);
+
+  const updateScroll = () => {
+    console.log('updateScroll');
+    if (ref.current && !isNil(scrollTop)) {
+      ref.current.scrollTop(scrollTop);
     }
-  }
+  };
 
-  updateAfterMount() {
-    if (this.ref && this.ref.current) {
-      const scrollbar = this.ref.current as any;
+  const updateAfterMount = () => {
+    if (ref && ref.current) {
+      const scrollbar = ref.current as any;
       if (scrollbar.update) {
         scrollbar.update();
       }
     }
```

Comment:
nit: Maybe using a quick exit here would make the code cleaner.
```suggestion
    if (!ref || !ref.current) {
      return;
    }

    const scrollbar = ref.current as any;
    if (scrollbar.update) {
      scrollbar.update();
    }
```